[Feat] add 'children' prop to Resources

hiaselhans opened the pull request:
React admin makes it fast to develop admin-interfaces with react.
However, when resources get more and more functions it is a bit unhandy to pass CustomRoutes to <Admin/>

This PR adds children as a property of resources and makes it possible to define the whole resource with custom sub-routes in a seperate typescript module:

``` tsx
import ListView from "./list";
import CustomRoute from "./customroute";

const MyResource = () => {
    return {
        list: ListView,
        children: <CustomRoute/>
    }
}
```

fzaninotto commented:
Thanks! Thats an interesting feature. 

As it's an enhancement, please rebase your PR against the `next` branch. 

Also, please document that feature and add some unit tests to make sure it works.

WiXSL commented on docs/CustomRoutes.md:
```diff
@@ -5,21 +5,24 @@ title: "The CustomRoutes Component"
 import { Admin, Resource, CustomRoutes } from 'react-admin';
 import { Route } from "react-router-dom";
-import posts from './posts';
+import posts, {PostAnalytics} from './posts';
```
```suggestion
import posts, { PostAnalytics } from './posts';
```

WiXSL commented on docs/CustomRoutes.md:
```diff
@@ -5,21 +5,24 @@ title: "The CustomRoutes Component"
     <Admin dataProvider={simpleRestProvider('http://path.to.my.api')}>
-        <Resource name="posts" {...posts} />
+        <Resource name="posts" {...posts}>
+            <Route path="analytics" element={<PostAnalytics/>}>
```
```suggestion
            <Route path="analytics" element={<PostAnalytics/>} />
```

WiXSL commented on docs/CustomRoutes.md:
```diff
@@ -5,21 +5,24 @@ title: "The CustomRoutes Component"
 // in src/App.js
 import * as React from "react";
 import { Admin, Resource, CustomRoutes } from 'react-admin';
 import { Route } from "react-router-dom";
```
```suggestion
import { Route } from "react-router-dom";
import simpleRestProvider from 'ra-data-simple-rest';
```

hiaselhans replied:
good catch!

WiXSL reviewed (changes requested):
Thanks for your work!
You are still missing some test to prove that it works.

hiaselhans commented:
thx @WiXSL !

i fixed the comments and added a test.
In my local env 3 unrelated tests are failing. However rebasing to next will flood the changeset in this PR

fzaninotto commented:
Thanks!

hiaselhans commented:
Great, thank you!
